src/main/java/Duke.java

```diff
@@ -2,93 +2,166 @@
+    public static final String LISTING_MESSAGE = "This are all the things I've remembered for you:";
+    public static final int VALID_ONE_PART_COMMAND = 1;
+    public static final int VALID_TWO_PART_COMMAND = 2;
+    public static final int MAX_TASK_LENGTH = 100;
 
-    public static void printLine() {
-        for(int i =0;i <= 50;i++) {
-            System.out.print("~");
-        }
-        System.out.println(" ");
-    }
-
-    public static void echoMessage(String line) {
-        System.out.println("Owl: I've remembered that!\nOwl: You said this: " + line);
-    }
-
-    public static void listTask(Task[] taskList) {
-        int taskIndex = 0;
-        int index = 1;
-        printLine();
-        System.out.println("This are all the things I've remembered for you:");
-        printLine();
-        for(Task task:taskList) {
-            Task theTask = taskList[taskIndex];
-            System.out.println(index + ".[" + (theTask.getStatus()) + "] "+ theTask.getDescription());
-            index++;
-            taskIndex++;
-        }
-        printLine();
-    }
 
     public static void main(String[] args) {
+        printWelcome();
+        Scanner in = new Scanner(System.in);
+        Task[] tasks = new Task[MAX_TASK_LENGTH];
+        int taskCount = 0;
+        String command = in.nextLine();
 
-        String LOGO = " ______   _       _   _\n"
-                + "|  __  | | | ___ | | | | \n"
-                + "| |  | | | |/   \\| | | | \n"
-                + "| |__| | |   / \\   | | |____\n"
-                + "|______| |__/   \\__| |______|\n";
-        System.out.println(LOGO);
-        printLine();
+        //while the command is not bye it keeps asking for more
+        commandHandler(in, tasks, taskCount, command);
+        printBye();
+    }
 
-        Scanner in = new Scanner(System.in);
+    /**
+     * Takes in commands from the command line interface and process it, different commands includes
+     * todo description which includes a todo task in the list marked with T
+     * deadline description /by when which includes a task with a specific deadline marked with D
+     * event description /at when to when includes a task with a specific timeframe marked with E
+     * list which list out all tasks in the list
+     * done taskIndex which checks the task as done with a "X"
+     * bye which terminates the OWL bot
+     *
+     * @param in Scanner object that takes in inputs from the command line interface
+     * @param tasks An array containing Todos, Deadline and Event objects
+     * @param taskCount The number of objects currently in the array
+     * @param command The given input from the user through the command liner interface
+     */
+    private static void commandHandler(Scanner in, Task[] tasks, int taskCount, String command) {
+        boolean isValid = false;
+        while(!command.equals("bye")) {
+            String[] brokenDownInput = command.split(" ", 2);
```

Comment:
brokenDownInput is a bit confusing. Can try to use name like "parameters" for context. Also the array should be plural.